api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel;
+
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.rel.partitions.Partition;
+
+public interface SupportsPartitions {
+
+  /**
+   * List all partition names of the table
+   *
+   * @return The list of partition names
+   */
+  String[] listPartitionNames();
+
+  /**
+   * List all partitions
+   *
+   * @return The list of partitions
+   */
+  Partition[] listPartitions();
+
+  /**
+   * Get a partition by partition name
```

Comment:
You'd better add more examples about how to get list partition or range partition as we discussed offline.

Reply:
added